tools/dia_dll.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+# Copyright (c) 2012 The Chromium Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+"""
+This script is based on chromium/chromium/master/tools/clang/scripts/update.py.
+
+It is used on Windows platforms to copy the correct msdia*.dll to the
+clang folder, as a "gclient hook".
+"""
+
+import os
+import shutil
+import stat
+import sys
+
+
+# Path constants. (All of these should be absolute paths.)
+THIS_DIR = os.path.abspath(os.path.dirname(__file__))
+LLVM_BUILD_DIR = os.path.abspath(os.path.join(THIS_DIR, '..', '..', 'third_party',
+                                              'llvm-build', 'Release+Asserts'))
+
+
+def GetDiaDll():
+  """Get the location of msdia*.dll for the platform."""
+
+  # Bump after VC updates.
+  DIA_DLL = {
+    '2013': 'msdia120.dll',
+    '2015': 'msdia140.dll',
+    '2017': 'msdia140.dll',
+    '2019': 'msdia140.dll',
+  }
+
+  # Don't let vs_toolchain overwrite our environment.
+  environ_bak = os.environ
+
+  sys.path.append(os.path.join(THIS_DIR, '..', '..', 'build'))
+  import vs_toolchain
+  win_sdk_dir = vs_toolchain.SetEnvironmentAndGetSDKDir()
+  msvs_version = vs_toolchain.GetVisualStudioVersion()
+
+  if bool(int(os.environ.get('DEPOT_TOOLS_WIN_TOOLCHAIN', '1'))):
+    dia_path = os.path.join(win_sdk_dir, '..', 'DIA SDK', 'bin', 'amd64')
+  else:
+    if 'GYP_MSVS_OVERRIDE_PATH' not in os.environ:
```

Comment:
Do we have existing use of this environment variable? If not, we should name it something else, rather than use a legacy name based on a build system that Flutter has never used.

Reply:
I believe this is used by depot_tools.  I'd rather avoid creating a new one that might not play well with the existing buildroot/depot_tools